src/validation.cpp

```diff
@@ -4129,7 +4153,12 @@ bool LoadExternalBlockFile(FILE* fileIn, CDiskBlockPos *dbp)
                     while (range.first != range.second) {
                         std::multimap<uint256, CDiskBlockPos>::iterator it = range.first;
                         std::shared_ptr<CBlock> pblockrecursive = std::make_shared<CBlock>();
-                        if (ReadBlockFromDisk(*pblockrecursive, it->second))
+
+                        BlockMap::iterator miSelf = mapBlockIndex.find(hash);
+                        CBlockIndex *pindex = nullptr;
+                        pindex = miSelf->second;
```

Comment:
this code could be unsafe, `miSelf` maybe null when the block is not in `mapBlockIndex`. 

since we have a block pointer here, can we try to read height from the block itself, `pblockrecursive->vtx[0]->vin[0].prevout.n`